source_gen/lib/src/builder.dart

```diff
@@ -186,6 +194,13 @@ class SharedPartBuilder extends _Builder {
 }
 
 /// A [Builder] which generates `part of` files.
+///
+/// Each output should correspond to a `part` directive in the primary input,
```

Comment:
We could consider stronger wording here, really nobody should ever use this builder any more.